### ISSUE TITLE
Forcefield.name attribute?

### ISSUE THREAD

**Issue by mattwthompson** [mosdef-hub/foyer#153]
In at least one of the atom typing errors, it is hard-coded to report that it found something wrong with the OPLS types whether or not we are using OPLS. For example:

```
AssertionError: Found inconsistent OPLS types in Generic title: [(10, 'mycoolff_044', 'mycoolff_045')]
```

I could simply change this to report `blah blah atom types ...` but maybe better would be to have it report the force field name. This would, however require adding a `name` attribute to the `Forcefield` class. I was thinking it would be optional, and would only take up one line of an xml. This should only require small changes.

Adding an attribute for a marginally cooler error message seems like overkill, but it seems natural to me that it should have one. We already can initiallize a `Forcefield` using a `name` argument (though this means something slightly different):

```
>>> OPLSAA = Forcefield(name='oplsaa')
KeyboardInterrupt
>>> OPLSAA.name
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Forcefield' object has no attribute 'name'

```

Thoughts?

**Comment by ctk3b** [comment]
Good catch. I think a first really simple PR will be to simply strike out the OPLS references [here](https://github.com/mosdef-hub/foyer/blob/b5bf96356c40069f110a907a749be6fb6ed62415/foyer/tests/utils.py#L30-L48). 

I like the name suggestion also but maybe let's do that in a separate PR since that requires deciding precisely how to name forcefields. Is simply going to be helpful metadata or something that we want to rely on to be precise? Does the .xml file name encode the forcefield name or do we add a separate xml tag?

**Comment by mattwthompson** [comment]
> Good catch. I think a first really simple PR will be to simply strike out the OPLS references here.

👍 Exactly what I was about to do.

> I like the name suggestion also but maybe let's do that in a separate PR since that requires deciding precisely how to name forcefields. Is simply going to be helpful metadata or something that we want to rely on to be precise?

My thinking was making this an optional argument and attribute would help us avoid some of these harder issues.

> Does the .xml file name encode the forcefield name or do we add a separate xml tag?

Except this as the immediately confusing one. I think the metadata in the xml should be considered more important, but we should also think about better ways to access force field files than their name. I'm starting to use multiple force fields simultaneously and while it's working well enough so far, accessing them by file name seems clunky to me.

**Comment by summeraz** [comment]
I think it makes the most sense to add an attribute to the `ForceField` tag within the XML that specifies the name, e.g. `<ForceField name="OPLS-AA">`. Then rather than Foyer looking for the name of forcefields minus the XML extension, it can just search for them based on the name in the XML (it should be easy enough to grab the XMLs within the "forcefields" directory in Foyer and compare their names with what the user provides). As far as user-defined force fields, I'm not sure there's a better way to load those than providing the filename, as the only other option I can think of would be to search through the user's entire path for force field XML files. Perhaps we could have a separation FOYERPATH environment variable that lists the paths to directories to search for force fields?

**Comment by ctk3b** [comment]
> it can just search for them based on the name in the XML (it should be easy enough to grab the XMLs within the "forcefields" directory in Foyer and compare their names with what the user provides). 

This may also be a none issue if/when we move all forcefields into their own repos. At that point, we would be deprecating the `Forcefield(name='foobar')` functionality since you'll have to either pass it a filename or something like a git commit hash.

> As far as user-defined force fields, I'm not sure there's a better way to load those than providing the filename, as the only other option I can think of would be to search through the user's entire path for force field XML files. Perhaps we could have a separate FOYERPATH environment variable that lists the paths to directories to search for force fields?

That would certainly work. Seems a bit like overkill but someone actually juggling lots of different forcefields like @mattwthompson would be better suited to comment on this. If/when forcefields become their own repos, you could just import them python style which IMO is much cleaner.

**Comment by mattwthompson** [comment]
Implemented #298
